crates/pico/src/derived_node.rs

```diff
@@ -48,3 +62,52 @@ pub struct DerivedNodeRevision {
     pub time_updated: Epoch,
     pub time_verified: Epoch,
 }
+
+#[derive(Debug, Clone, Copy)]
+pub struct MemoRef<'db, T> {
+    db: &'db Database,
+    derived_node_id: DerivedNodeId,
+    phantom: PhantomData<T>,
+}
+
+impl<'db, T: 'static + Clone> MemoRef<'db, T> {
```

Comment:
is everything going to be `T: 'static`?

For example, what if we have
```
#[memo]
fn get_client_field() -> ClientField

#[memo]
fn get_client_field_selections() -> &ClientFieldSelections
```

Returning a subset or view into another object will probably be a common pattern, because this is what will allow us to avoid unnecessary recalculation. e.g. if the AST changed, but not in a way that affected the selections, we would need something like this to avoid doing a bunch of extra work

If this is the case, are we forced to keep clones of the `selection_set` redundantly in the database?